be/src/fs/fs_starlet.cpp

```diff
@@ -232,13 +216,14 @@ class StarletFileSystem : public FileSystem {
             return Status::NotSupported(fmt::format("Starlet: cannot create file with name ended with '/': {}", path));
         }
 
-        ASSIGN_OR_RETURN(auto conf, get_shard_config(pair.second))
-
-        auto fs_st = FileSystemFactory::new_filesystem(conf["scheme"], conf);
+        auto fs_st = get_shard_filesystem(pair.second);
```

Comment:
why not use the existed macro `RETURN_IF_ERROR` ?

Reply:
starlet library uses absl::StatusOr which is not the same as starrocks::StatusOr, so RETURN_IF_ERROR doesn't fit.